src/util/config.js

```diff
@@ -1,97 +1,183 @@
-			'https://secure.dev.meetup.com/oauth2/authorize',
-		OAUTH_ACCESS_URL: process.env.OAUTH_ACCESS_URL ||
-			process.env.ANONYMOUS_ACCESS_URL ||
-			'https://secure.dev.meetup.com/oauth2/access',
-		PHOTO_SCALER_SALT: process.env.PHOTO_SCALER_SALT,
-		oauth: {
-			secret: process.env.MUPWEB_OAUTH_SECRET,
-			key: process.env.MUPWEB_OAUTH_KEY,
-		},
-	};
-	config.duotoneUrls = getDuotoneUrls(duotones, config.PHOTO_SCALER_SALT);
-	config.API_SERVER_ROOT_URL = `${config.API_PROTOCOL}://${config.API_HOST}`;
+};
 
-	// currently all config is available syncronously, so resolve immediately
-	return Promise.resolve({ ...config, ...overrideConfig }).then(validateConfig);
-}
+export const validateOauthSecret = secret => {
+	if (!secret || secret.toString().length < 1) {
+		throw new Error(OAUTH_SECRET_ERROR);
+	}
+};
+
+export const validateOauthKey = key => {
+	if (!key || key.toString().length < 1) {
+		throw new Error(OAUTH_KEY_ERROR);
+	}
+};
 
-function validateConfig(config) {
-	const oauthError = new Error('get oauth secrets from web platform team');
-	const configSchema = Joi.object()
-		.keys({
-			API_PROTOCOL: Joi.any().only(['https', 'http']).required(),
-			API_HOST: Joi.string().hostname().required(),
-			API_SERVER_ROOT_URL: Joi.string().uri(),
-			API_TIMEOUT: Joi.number(),
-			COOKIE_ENCRYPT_SECRET: Joi.string()
-				.min(32)
-				.required()
-				.error(
-					new Error(
-						'set COOKIE_ENCRYPT_SECRET env variable to a random 32+ character string'
-					)
-				),
-			CSRF_SECRET: Joi.string()
-				.min(32)
-				.required()
-				.error(
-					new Error(
-						'set CSRF_SECRET env variable to a random 32+ character string'
-					)
-				),
-			DEV_SERVER_PORT: Joi.number().integer().max(65535),
-			OAUTH_AUTH_URL: Joi.string().uri().required(),
-			OAUTH_ACCESS_URL: Joi.string().uri().required(),
-			PHOTO_SCALER_SALT: Joi.string()
-				.min(1)
-				.required()
-				.error(new Error('get PHOTO_SCALER_SALT from web platform team')),
-			oauth: Joi.object()
-				.keys({
-					secret: Joi.string().min(1).required().error(oauthError),
-					key: Joi.string().min(1).required().error(oauthError),
-				})
-				.required(),
-			duotoneUrls: Joi.object(),
-		})
-		.required();
-
-	const result = Joi.validate(config, configSchema);
-	if (result.error) {
-		throw result.error;
+export const validatePhotoScalerSalt = salt => {
+	if (!salt || salt.toString().length < 1) {
+		throw new Error(SALT_ERROR);
 	}
-	return result.value;
+};
+
+let config = convict({
+	env: {
+		format: ['production', 'development', 'test'],
+		default: 'development',
+		env: 'NODE_ENV',
+	},
+	api: {
+		protocol: {
+			format: validateProtocol,
+			default: 'https',
+			env: 'API_PROTOCOL',
+		},
+		host: {
+			format: String,
+			default: 'api.dev.meetup.com',
+			env: 'API_HOST',
+		},
+		timeout: {
+			format: 'int',
+			default: 8000,
+			env: 'API_TIMEOUT',
+		},
+		root_url: {
+			format: String,
+			default: '',
+		},
+	},
+	asset_server: {
```

Comment:
Just thought of this - 

`asset_server` is a `mup-web` configuration, and not related to the platform code. Should mup-web have it's own convict instance for `asset_server`, `asset_path`, `enable_hmr`, and other such consumer configurations?